src/pudl/analysis/allocate_gen_fuel.py

```diff
@@ -849,13 +849,27 @@ def identify_retiring_generators(gen_assoc: pd.DataFrame) -> pd.DataFrame:
     report generator-specific generation data in the g table after their retirement
     date.
     """
-    retiring_generators = gen_assoc.loc[
+    # identify the complete set of generator ids that match this criteria
+    retiring_generator_identities = gen_assoc.loc[
         (gen_assoc.operational_status == "retired")
         & (
             (gen_assoc.report_date <= gen_assoc.generator_retirement_date)
             | (gen_assoc.net_generation_mwh_g_tbl.notnull())
-        )
-    ]
+        ),
+        ["plant_id_eia", "generator_id"],
+    ].drop_duplicates()
+
+    # merge these ids into gen_assoc and keep all months of data for these gens
+    retiring_generators = gen_assoc.copy().merge(
+        retiring_generator_identities,
+        how="outer",
+        on=["plant_id_eia", "generator_id"],
+        indicator="_retiring_gens",
+    )
+
+    retiring_generators = retiring_generators[
+        retiring_generators["_retiring_gens"] == "both"
+    ].drop(columns="_retiring_gens")
```

Comment:
It does not really matter but can we do this with an **inner** join (intersection)? So we would do this:
```
retiring_generators = gen_assoc.copy().merge(retiring_generator_identities, how="inner", on=["plant_id_eia", "generator_id"])
```
It should be equivqlent than doing the union (`how="outer"`) and then keeping only rows that appears in both tables (indicator is `both`)

Reply:
That could work too, this is just the convention that was used by PUDL, and since we may have to eventually merge this back into pudl I was just trying to keep this consistent.